Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -99,11 +97,35 @@ function Get-DiagnosticsExtensions($storageAccount, $extensionsPath)
                         $publicConfigStorageAccountName = $publicConfig.PublicConfig.StorageAccount
                         Write-Verbose "Found PublicConfig.StorageAccount= '$publicConfigStorageAccountName'"
 
-                        $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName
+                        try
+                        {
+                            $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName
+                        }
+                        catch
+                        {   
+                            Write-Host (Get-VstsLocString -Key "Unabletofind0usingprovidedsubscriptionLookinginstoragekeys" -ArgumentList "$publicConfigStorageAccountName")
+                            Write-Verbose $_.Exception.Message
+                        }
                         if ($publicConfigStorageKey)
                         {
-                            Write-Verbose "##[command]New-AzureStorageContext -StorageAccountName $publicConfigStorageAccountName -StorageAccountKey <key>"
-                            $storageContext = New-AzureStorageContext -StorageAccountName $publicConfigStorageAccountName -StorageAccountKey $publicConfigStorageKey.Primary
+                            Write-Verbose "##Getting storage account name and key from diagnostics config file"
+
+                            Write-Verbose "##$storageAccountName = $publicConfigStorageAccountName"
+                            $storageAccountName = $publicConfigStorageAccountName
+
+                            Write-Verbose "##$storageAccountKey = $publicConfigStorageKey.Primary"
+                            $storageAccountKey = $publicConfigStorageKey.Primary
+                        }                    
+                        elseif($storageAccountKeysMap.containsKey($role))
```

Comment:
If a storageAccountKeyMap is given, then we should honor that first. Can you move this logic up? We should do following:
1. If there is $publicConfig.PublicConfig.StorageAccount, and if storageAccountKeysMap is available, then you should look at that first.
2. If #1 is not true, then look for key from diagnostics config file
3. If #2 is not true, then attempt to get key using Get-AzureStorageKey